### ISSUE TITLE
Make it easy to disable Immutable***Rules

### ISSUE THREAD

**Issue by atkawa7** [PicnicSupermarket/error-prone-support#313]
### Describe the bug
We currently do not have a way to disable these rules as we do not use guava on our codebase. Is there a way to disable these globaly without using annotations `SuppressWarnings(...)`. An example that triggers the warning. 
```java
var array  = List.of(....)
```

**Comment by rickie** [comment]
Thanks for reaching out! 

Yes we definitely provide a way to disable specific Refaster rules. Using a regex one can restrict which Refaster rules will be loaded. For some extra context, see [the explanation in `Refaster.java`](https://github.com/PicnicSupermarket/error-prone-support/blob/master/refaster-runner/src/main/java/tech/picnic/errorprone/refaster/runner/Refaster.java#L50). 

Let me provide a solution for which I verified with that your provided example is not flagged anymore. 
Please add this to your Error Prone configuration in the `pom.xml`: 
```
-XepOpt:Refaster:NamePattern=^(?!(Immutable.*Rules)).*
```
This enables all rules except for the rule collections that match `Immutable.*Rules`. 

Does this help you @atkawa7 ?

**Comment by atkawa7** [comment]
Thanks @rickie. That solved the issue. Closing this

**Comment by Stephan202** [comment]
First off: congrats @atkawa7 on opening the _very first_ issue in this repository :smile:.

For others landing here: we are aware that we can do better in this respect:
1. For `BugChecker`s _other than_ `Refaster` we have #308, which should prevent Error Prone Support (by default) from introducing a dependency on libraries not already on the classpath. The plan is to do something similar for the `Refaster` check (but that'll require a significantly different approach; details TBD).
2. We should also better document `-XepOpt:Refaster:NamePattern` and other custom flags :)

I'll link the relevant PRs here if/when they land.
